x-pack/plugins/observability_solution/synthetics/server/routes/overview_trends/fetch_trends.ts

```diff
@@ -0,0 +1,143 @@
+
+import { SearchRequest } from '@elastic/elasticsearch/lib/api/types';
+import { SyntheticsEsClient } from '../../lib';
+
+export function fetchTrends(configId: string, locationIds: string[], esClient: SyntheticsEsClient) {
+  const query: SearchRequest = {
+    size: 0,
+    query: {
+      bool: {
+        filter: [
+          {
+            bool: {
+              filter: [
+                {
+                  exists: {
+                    field: 'summary',
+                  },
+                },
+                {
+                  bool: {
+                    should: [
+                      {
+                        bool: {
+                          should: [
+                            {
+                              match: {
+                                'summary.final_attempt': true,
+                              },
+                            },
+                          ],
+                          minimum_should_match: 1,
+                        },
+                      },
+                      {
+                        bool: {
+                          must_not: {
+                            bool: {
+                              should: [
+                                {
+                                  exists: {
+                                    field: 'summary.final_attempt',
+                                  },
+                                },
+                              ],
+                              minimum_should_match: 1,
+                            },
+                          },
+                        },
+                      },
+                    ],
+                    minimum_should_match: 1,
+                  },
+                },
+              ],
+            },
+          },
+          {
+            bool: {
+              must_not: {
+                exists: {
+                  field: 'run_once',
+                },
+              },
+            },
+          },
+          {
+            terms: {
+              'observer.name': locationIds,
+            },
+          },
+          {
+            term: {
+              config_id: configId,
+            },
+          },
+          {
+            range: {
+              '@timestamp': {
+                gte: 'now-9h',
+                lte: 'now',
+              },
+            },
+          },
+        ],
+      },
+    },
+    aggs: {
+      byId: {
+        terms: {
+          field: 'config_id',
+        },
+        aggs: {
+          byLocation: {
+            terms: {
+              field: 'observer.name',
+            },
+            aggs: {
+              last_50: {
+                top_hits: {
+                  size: 50,
+                  sort: [
+                    {
+                      '@timestamp': {
+                        order: 'desc',
+                      },
+                    },
+                  ],
+                  _source: ['monitor.duration.us'],
+                },
+              },
```

Comment:
we can replace this with `top_metrics`

Reply:
`top_metrics` has a `size` cap of 10, so this would result in a different user-facing behavior of the trend visualization.

From the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-top-metrics.html#_size_2):

```
The default size is 1. The maximum default size is 10 because the aggregation’s
working storage is "dense", meaning we allocate size slots for every bucket.
10 is a very conservative default maximum and you can raise it if you need to
by changing the top_metrics_max_size index setting. But know that large sizes
can take a fair bit of memory, especially if they are inside of an aggregation
which makes many buckes like a large terms aggregation.
```

IMO it's ok to just use `top_hits` and take the perf hit vs. adding this deviation from defaults; if we want to do this I think it is better to do it in a dedicated follow-up where the sole attention of the patch is modifying this setting.

Reply:
I think in that case, we can just histogram here, maybe, I just think top_hits is too expensive for this use case

Reply:
I'll experiment with histogram today. My only concern there is usually bucket overflow but since we are scoping to a single `monitorId` * `numLocation` it's probably alright, and I agree histogram will be more performant for sure.